lib/controllers/history_controller.dart

```diff
@@ -0,0 +1,113 @@
+import 'dart:collection';
+
+import 'package:flutter/material.dart';
+import 'package:history_manager/history_manager.dart';
+
+/// - Holds undo and redo functionalities for the session of this controller.
+/// - Extends to [ChangeNotifier] to notify changes for the mutable properties to take changes in the UI State.
+class HistoryController extends ChangeNotifier {
+  HistoryController();
+
+  // * Mutable Properties
+  /// Stack of executed changes during a session.
+  Queue<ExecutedChange> _executedChanges = ListQueue();
```

Comment:
Allocate some initial capacity so it's possible to keep append operations at O(1). Maybe 10 is good enough

Reply:
This is also why appending should be preferred over adding in the 0 index. Elements probably need to be moved. With appending, this doesn't happen

Reply:
![image](https://user-images.githubusercontent.com/62531088/170059144-2c5be757-81af-41d8-9abe-aa9ea53f2650.png)

by default tho it has an initial capacity of 8, should I retain as is for the moment or we go with 10?

Reply:
Hmm maybe the best way is to allow the developer to set what initial capacity they want for both ListQueues at initialization.

Reply:
okay, let's have that passed but I'll make it optional. when nothing is passed let's have it at 10?

Reply:
Is there no concept of default arguments in Dart?

Reply:
Not sure how it works in Dart but if the initializer is defined as follows:

struct Person(name: String = "John Doe")

let x = Person() // this should work

Wouldn't this apply here?